<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Exercise-9-10.1" data-toc-modified-id="Exercise-9-10.1-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Exercise 9-10.1</a></span><ul class="toc-item"><li><span><a href="#Question-1" data-toc-modified-id="Question-1-1.1"><span class="toc-item-num">1.1&nbsp;&nbsp;</span>Question 1</a></span></li><li><span><a href="#Question-2" data-toc-modified-id="Question-2-1.2"><span class="toc-item-num">1.2&nbsp;&nbsp;</span>Question 2</a></span></li></ul></li><li><span><a href="#Exercise-9-10.2" data-toc-modified-id="Exercise-9-10.2-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>Exercise 9-10.2</a></span><ul class="toc-item"><li><span><a href="#Question-1" data-toc-modified-id="Question-1-2.1"><span class="toc-item-num">2.1&nbsp;&nbsp;</span>Question 1</a></span></li><li><span><a href="#Question-2" data-toc-modified-id="Question-2-2.2"><span class="toc-item-num">2.2&nbsp;&nbsp;</span>Question 2</a></span></li><li><span><a href="#Question-3" data-toc-modified-id="Question-3-2.3"><span class="toc-item-num">2.3&nbsp;&nbsp;</span>Question 3</a></span></li><li><span><a href="#Question-4" data-toc-modified-id="Question-4-2.4"><span class="toc-item-num">2.4&nbsp;&nbsp;</span>Question 4</a></span></li></ul></li><li><span><a href="#Exercise-9-10.3" data-toc-modified-id="Exercise-9-10.3-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>Exercise 9-10.3</a></span><ul class="toc-item"><li><span><a href="#Question-1" data-toc-modified-id="Question-1-3.1"><span class="toc-item-num">3.1&nbsp;&nbsp;</span>Question 1</a></span></li><li><span><a href="#Question-2" data-toc-modified-id="Question-2-3.2"><span class="toc-item-num">3.2&nbsp;&nbsp;</span>Question 2</a></span></li><li><span><a href="#Question-3:-Interpret-the-results-obtained-above" data-toc-modified-id="Question-3:-Interpret-the-results-obtained-above-3.3"><span class="toc-item-num">3.3&nbsp;&nbsp;</span>Question 3: Interpret the results obtained above</a></span></li><li><span><a href="#Question-4:" data-toc-modified-id="Question-4:-3.4"><span class="toc-item-num">3.4&nbsp;&nbsp;</span>Question 4:</a></span></li><li><span><a href="#Question-5:" data-toc-modified-id="Question-5:-3.5"><span class="toc-item-num">3.5&nbsp;&nbsp;</span>Question 5:</a></span></li></ul></li><li><span><a href="#Exercise-9-10.3" data-toc-modified-id="Exercise-9-10.3-4"><span class="toc-item-num">4&nbsp;&nbsp;</span>Exercise 9-10.3</a></span><ul class="toc-item"><li><span><a href="#Question-1" data-toc-modified-id="Question-1-4.1"><span class="toc-item-num">4.1&nbsp;&nbsp;</span>Question 1</a></span></li><li><span><a href="#Question-2" data-toc-modified-id="Question-2-4.2"><span class="toc-item-num">4.2&nbsp;&nbsp;</span>Question 2</a></span></li><li><span><a href="#Question-3" data-toc-modified-id="Question-3-4.3"><span class="toc-item-num">4.3&nbsp;&nbsp;</span>Question 3</a></span></li><li><span><a href="#Question-4" data-toc-modified-id="Question-4-4.4"><span class="toc-item-num">4.4&nbsp;&nbsp;</span>Question 4</a></span></li></ul></li></ul></div>

In [1]:
import numpy as np
import sympy as sp
import pandas as pd
import matplotlib.pyplot as plt
from scipy import stats, optimize
import seaborn as sns

config = {
    'figure.facecolor':'w',
    'axes.grid':True,
    'axes.spines.top':False,
    'axes.spines.bottom':False,
    'axes.spines.left':False,
    'axes.spines.right':False,
    'grid.linewidth':0.5,
    'grid.linestyle':'--',
    'figure.constrained_layout.use':True
}

plt.rcParams.update(config)

# Exercise 9-10.1

In 1985, the yield associated to US short term Treasury Notes was about 6%. Assuming that he market expected return is 30% for a portfolio with a beta of 2.

## Question 1

According to the CAPM what is the expected return for the market portfolio?

The CAPM says:

$$\mathbb E[r_i] = r_f + \beta \mathbb E[r_m - r_f]$$

We're told that "a portfolo" has an expected return of 30%, and it has a $\beta$ of 2. We're asked to do some algebra and solve for $\mathbb E[r_m]$:

In [1]:
r_f = 0.06
r_i = 0.3
β = 2
print(f'CAPM estimated market return: {(r_i - r_f) / β + r_f :0.2%}')

CAPM estimated market return: 18.00%


## Question 2

The stock Astrotech has been sold 40\\$. An analyst forecast a dividend of 3\\$ in 1988 and an ask price of 41\\$. If the forecast of the analyst are true and β = −0.4, was the stock over or under priced?

First, compute the expected price under the CAPM:

In [8]:
r_i = r_f + -0.4 * 0.12
r_i

0.011999999999999997

This will be the expected increase in the stock price to next year. To tie this into the price, we can think about the return as the percent change in the stock price, including the dividend. That is:

$$r_i = \frac{P_1 + D_1}{P_0} - 1$$

Note that this is just percent change from $t=0$ to $t=1$, adjusting for the dividend payout. We need to solve for $P_0$:

$$P_0 = \frac{P_1 + D_1}{1 + r_i}$$

In [15]:
print(f'Implied price at P0: {(41 + 3) / (1 + r_i):0.2f}$')

Implied price at P0: 43.48$


Since the implied price of 43.48 is greater than the actual asking price of 41, we conclude that the stock was under-priced.

# Exercise 9-10.2

The risk-free asset return is 2.5%. You can invest in a private equity firm whose return and
associated probability are presented in the following table:

|              |   1  |   2  |
|--------------|------|------|
| Return       | +25% | -50% |
| Probability  | 70%  | 30%  |

## Question 1

Compute the variance and the expected return of the investment

In [21]:
r_f = 0.025
p = np.array([0.7, 0.3])
r = np.array([0.25, -0.5])
μ = (r * p).sum()
σ2 = (r ** 2 * p).sum() - μ ** 2
print(f'Expected return: {μ:0.3f}')
print(f'Variance: {σ2:0.3f}')
print(f'Volatility: {np.sqrt(σ2):0.3f}')

Expected return: 0.025
Variance: 0.118
Volatility: 0.344


## Question 2

Assuming that your behaviour can be modelled by a mean-variance utility function, and that you do not have any extra information about this investment, which part of you wealth could you invest in?

We're goning to make a portfolio between this asset and the risk-free asset. Call the weight on the risky asset $w$, then we will have:

$$\begin{align}\mu_p &= w\mu + (1 - w) r_f \\
    \sigma_p^2 &= w^2 \sigma^2 \end{align}$$
    
So utility is:

$$\begin{align}U &= \mu_p - \frac{A}{2}\sigma_p^2 \\
                 &= w\mu_p + (1 - w)r_f - \frac{A}{2}w^2 \sigma^2 \end{align}$$
                 
Take derivative and solve:

$$w = \frac{\mu - r_f}{A\sigma^2_p}$$

We don't know our $A$, but it doesn't matter! Notice that $r_f = 2.5\%$, and we computed $\mu_p = 2.5\%$. So no matter what the risk tolerance, we choose $w=0$ -- we don't want this portfolio at all!

## Question 3

Now we are told that:

- Expected Market Return $E[R_m]$ = 6.5%
- Market Variance $V[R_m]$ = 0.0289
- Covariance between the market and the asset $\sigma_{iM}$ = -0.00289

From this information, we can compute $\beta = \frac{\sigma_{i,M}}{\sigma^2_M}$:

In [23]:
σ_im = -0.00289
σ2_m = 0.0289
μ_m = 0.065

β = σ_im / σ2_m
print(f'CAPM beta for this asset: {β:0.3f}')

CAPM beta for this asset: -0.100


And we can compute the expected returns of this asset, given this beta with the market:
$$\mathbb E [r_i] = r_f + \beta (\mathbb E[r_m] - r_f)$$

In [26]:
Er_i = r_f + β * (μ_m - r_f)
Er_i

0.021

Under the CAPM, we compute that the expected return is 2.1%. Notice that this is lower than the observed return of 2.5%. This suggests that the asset is underpriced, so it is rational to buy it. This is despite the fact that, in any case, the return is lower than the risk-free rate. 

## Question 4

Determine the threshold $\sigma_{i,M}$ from which you will stop investing in this private equity firm.

We will stop if $\mathbb E[r_i] = r_i = 0.025$, so we just solve for $\sigma_{i,M}$ such that:

$$0.025 = r_f + \frac{\sigma_{i,M}}{\sigma^2_M}(\mu_m - r_f)$$

In [30]:
σ_im_stop = (0.025 - r_f) * σ2_m / (μ_m - r_f)
print(f'σ_im at which we stop investing: {σ_im_stop:0.3f}')

σ_im at which we stop investing: 0.000


How should be interpret this? Remember that this is the **covariance**, not the variance, so we shouldn't be shocked that it's zero. Remember that the beta with the market was negative. Basically, we like this asset because it provides a downside hedge against market declines -- when the market goes down, this asset goes up. If it stops serving that function -- if the beta becomes positive -- we stop being interested in this asset.

# Exercise 9-10.3

The asset price at date 0 is denoted $P_0$. The asset bid price at date 1 is a random variable denoted $P_1$. Assuming the CAPM framework is valid and that the beta of the asset is denoted $\beta$.

## Question 1

Prove that $P_0 = \frac{E(\tilde{P}_1)}{(1 + r_f + (\mu_M - r_f)\beta)}$

First, we note that $\mathbb E[P_1] = (1 + \mathbb E[r_1])P_0$. Solving for $P_0$, we get:

$$P_0 = \frac{\mathbb E[P_1]}{(1 + \mathbb E[r_1])}$$

Great, we're already most of the way there. Now we just need to know what is $\mathbb E[r_1]$. We're told that the CAPM framework is valid, and that tells us that:

$$\mathbb E[r_1] = r_f + \beta (\mathbb E[r_m] - r_f)$$

Define $\mu_m = \mathbb E[r_m]$, then just plug this expression into our fraction:

$$P_0 = \frac{\mathbb E[P_1]}{(1 + r_f + \beta (\mu_m - r_f))}$$

The expression is verified.

## Question 2

Show that we can re-write the above expression to:

$$P_0 = \frac{\mathbb E[P_1] - a \rho \sigma}{1 + r_f}$$

**Important note**: There is a typo in the English version of the question! The english version says that $\rho$ is the correlation between the returns of the asset and the market. But the French version says it is the correlation between the **PRICE** of the asset and the returns of the market. This is a very important distinction, as we will see.

Where $\rho$ is the correlation between the **PRICE OF THE ASSET** and the market **RETURN**, $\sigma$ is the voltility of the asset, and $a$ is an unknown parameter.

Let's start from the CAPM expression:

$$\mathbb E[r_1] = r_f + \beta (\mathbb E[r_m] - r_f)$$

We know that we can write $\beta = \frac{\sigma_{i,m}}{\sigma^2_m}$, because we can think about the CAPM as a linear regression.

Plug this into the percent change equation:

$$P_0 = \frac{\mathbb E[P_1]}{1 + r_f + \frac{\sigma_{i,m}}{\sigma^2_m} (\mu_m - r_f)}$$

Define $S_m = \frac{\mu_m - r_f}{\sigma_m}$ as the Sharpe ratio of the market. This gives:

$$P_0 = \frac{\mathbb E[P_1]}{1 + r_f + S_m \frac{\sigma_{i,m}}{\sigma_m}}$$

Now shuffle some stuff around:

$$\begin{align}
P_0 \left (1 + r_f + S_m \frac{\sigma_{i,m}}{\sigma_m} \right ) &= \mathbb E[P_1] \\
(1 + r_f) P_0 + P_0 S_m \frac{\sigma_{i,m}}{\sigma_m} &= \mathbb E[P_1] \\
\end{align}$$

We really want to get rid of the $P_0$ in the 2nd summand on the left-hand side. If we could, we would be done. The final twist in the derivation will be to look inside $sigma_{i,m}$ for some help. We know that:

$$\begin{align}
\sigma_{i, m} &= \text{Cov}(r_i, r_m) \\ 
&= \text{Cov}\left (\frac{P_1}{P_0} - 1, r_m \right )
&= \text{Cov}\left (\frac{P_1}{P_0} - \frac{P_0}{P_0}, r_m \right )
\end{align}$$

These re-writes just use the definition of returns, plus a "clever" re-writing of the number 1. Next, standing here at time $t=0$, $P_0$ is non-random, so we can factor it out:

$$\sigma_{i, m} = \frac{1}{P_0} \text{Cov}\left (P_1 - P_0, r_m \right )$$

$\text{Cov}(P_1 - P_0, r_m)$ is actually equal to $\text{Cov}(P_1, r_m)$. You can convince yourself of this using the following argument:

1. Subtracting a deterministic number from a distribution shifts all values of that distribution equally
2. Therefore, the relative distances between the points does not change
3. Variance (and therefore covaraince) measures relative distance
4. Therefore, a shift does not change the variance.

Or, you can grind out the following algebra:

$$\begin{align}
\text{Cov}(P_1 - P_0, r_m) &= \mathbb E[((P_1 - P_0) - \mathbb E[(P_1 - P_0)])(r_m - \mathbb E[r_m])] \\
 &= \mathbb E[(P_1 - \mathbb E[P_1] - P_0 - \mathbb E[P_0])(r_m - \mathbb E[r_m])] \\
 &= \mathbb E[(P_1 - \mathbb E[P_1])(r_m - \mathbb E[r_m])] \\
 &= \text{Cov}(P_1, r_m)
\end{align}$$

Ok so we have:

$$\sigma_{i, m} = \frac{1}{P_0}  \text{Cov}(P_1, r_m)$$

Plug this into the equation we were working on:

$$\begin{align}
(1 + r_f) P_0 + P_0 S_m \frac{1}{P_0} \frac{\text{Cov}(P_1, r_m)}{\sigma_m} &= \mathbb E[P_1] \\
(1 + r_f) P_0 + S_m \frac{\text{Cov}(P_1, r_m)}{\sigma_m} &= \mathbb E[P_1] \\
\end{align}
$$

This is starting to look like what we want. Isolate $P_0$:

$$P_0 = \frac{\mathbb E[P_1] - S_m \frac{\text{Cov}(P_1, r_m)}{\sigma_m}}{1 + r_f}$$

We want the numerator on the right-hand side to look like $\mathbb E[P_1] - a \rho \sigma$. Define $a = S_m$, so we have that. Let's make a $\sigma_i$ appear by multiplying by $\frac{\sigma_i}{\sigma_i}$:

$$P_0 = \frac{\mathbb E[P_1] - a \frac{\text{Cov}(P_1, r_m)\sigma_i}{\sigma_m \sigma_i}}{1 + r_f}$$

Why do this sigma thing? Well, because now we have $\sigma_i \frac{\text{Cov}(P_1, r_m)}{\sigma_m \sigma_i}$, which **ALMOST** looks like the correlation coefficient between $P_1$ and $r_m$ -- $\text{Corr}(P_1, r_m) = \frac{\text{Cov}(P_1, r_m)}{\sigma_m \sigma_{P_i}}$. But... the question ignores this point, and just defines:

$$\rho_{i,m} = \frac{\text{Cov}(P_1, r_m)}{\sigma_m \sigma_i}$$

Doing this move, we get:

$$P_0 = \frac{\mathbb E[P_1] - a \rho_{i,m} \sigma_i}{1 + r_f}$$

With $a = S_m$ and $\rho_{i,m}$ defined as above. This verifies the expression.

## Question 3: Interpret the results obtained above

We have derived that the current price of an asset is equal to the expected present-discounted future value of that asset, $\frac{\mathbb E[P_1]}{1 + r_f}$, adjusted by a factor $- \frac{a \rho_{i,m}\sigma_i}{1 + r_f}$.

The adjustment factor is built from 1) the correlation between the price and the market return, 2) the variance of the returns on the asset, and 3) the Sharpe ratio of the asset. The factor is related to the **risk** of the asset, scaled by the correlation with the market. This is the risk-premium, so the quantity $\mathbb E[P_1] - a \rho_{i,m} \sigma_i$ can be thought of as the "risk-neutral expected price" of the asset. 

Also note that the risk premium is *subtracted*, even though it includes the correlation between the price and the market return. This suggets that we assume $\rho_{i,m} > 0$. It also means that assets with **negative** beta demand a higher price, because we add their risk premium to the price rather than subtract it.

## Question 4:

Now assume that the asset under consideration pays a dividend $\tilde X$ in all future periods. Re-do question 1 taking into account this new setup.

For assets with dividends, we have to add the dividend payment from the price before computing returns. That is:

$$r_t = \frac{P_t + \tilde X}{P_{t-1}} - 1$$

Evaluate this at $t=1$, conditioned on the $t=0$ information set:

$$\mathbb E[r_1] = \frac{\mathbb E[P_1] + \tilde X}{P_0} - 1$$

We re-arrange to solve for $P_0$:
$$
\begin{align}
1 + \mathbb E[r_1] &= \frac{\mathbb E[P_1] + \tilde X}{P_0} \\
P_0 &= \frac{\mathbb E[P_1] + \tilde X}{1 + \mathbb E[r_1]}
\end{align}
$$

From here, the steps are basically the same. Invoke the CAPM:

$$P_0 = \frac{\mathbb E[P_1] + \tilde X}{1 + r_f + \beta (\mu_m - r_f))}$$

We find that the expression is identical, but just adjusts for the dividend payment in the numerator.

## Question 5: 

Express the initial price of the asset in terms of expected future cash flows.

For this question, we do all the steps we just did, but don't substiute the CAPM. When you read "cash flows", you should think "infinite sum". Since we want the initial price, start from here:

$$P_0 = \frac{\mathbb E[P_1] + \tilde X}{1 + \mathbb E[r_1]}$$

This is true in all time periods, so we can also write:

$$\mathbb E[P_1] = \frac{\mathbb E[P_2] + \tilde X}{1 + \mathbb E[r_2]}$$

I want to plug this into the first equation, but first I want to split the addition in the numerator to make the substitution a bit easier to work with:

$$\begin{align}
P_0 &= \frac{\mathbb E[P_1]}{1 + \mathbb E[r_1]} + \frac{\tilde X}{1 + \mathbb E[r_1]} \\
P_0 &= \frac{1}{1 + \mathbb E[r_1]} \mathbb E[P_1] + \frac{\tilde X}{1 + \mathbb E[r_1]} \\
P_0 &= \frac{1}{1 + \mathbb E[r_1]} \left ( \frac{\mathbb E[P_2] + \tilde X}{1 + \mathbb E[r_2]} \right ) + \frac{\tilde X}{1 + \mathbb E[r_1]} \\
\end{align}$$

Do some clean-up...
$$\begin{align}
P_0 &= \frac{\mathbb E[P_2]}{(1 + \mathbb E[r_1])(1 + \mathbb E[r_2])} + \frac{\tilde X}{1 + \mathbb E[r_1]} + \frac{\tilde X}{(1 + \mathbb E[r_1])(1 + \mathbb E[r_2])} \\
\end{align}$$

Maybe you see the pattern? Let's do one more. We can also write an expression for $\mathbb E[P_2]$ by shifting the time indices around:

$$\mathbb E[P_2] = \frac{\mathbb E[P_3] + \tilde X}{1 + \mathbb E[r_3]}$$

Substitute this in:
$$\begin{align}
P_0 &= \frac{\mathbb E[P_2]}{(1 + \mathbb E[r_1])(1 + \mathbb E[r_2])} + \frac{\tilde X}{1 + \mathbb E[r_1]} + \frac{\tilde X}{(1 + \mathbb E[r_1])(1 + \mathbb E[r_2])} \\
P_0 &= \frac{1}{(1 + \mathbb E[r_1])(1 + \mathbb E[r_2])}\mathbb E[P_2] + \frac{\tilde X}{1 + \mathbb E[r_1]} + \frac{\tilde X}{(1 + \mathbb E[r_1])(1 + \mathbb E[r_2])} \\
P_0 &= \frac{1}{(1 + \mathbb E[r_1])(1 + \mathbb E[r_2])} \left ( \frac{\mathbb E[P_3] + \tilde X}{1 + \mathbb E[r_3]} \right ) + \frac{\tilde X}{1 + \mathbb E[r_1]} + \frac{\tilde X}{(1 + \mathbb E[r_1])(1 + \mathbb E[r_2])} \\
P_0 &= \frac{\mathbb E[P_3]}{(1 + \mathbb E[r_1])(1 + \mathbb E[r_2])(1 + \mathbb E[r_3])} + \frac{\tilde X}{1 + \mathbb E[r_1]} + \frac{\tilde X}{(1 + \mathbb E[r_1])(1 + \mathbb E[r_2])} + \frac{\tilde X}{(1 + \mathbb E[r_1])(1 + \mathbb E[r_2])(1 + \mathbb E[r_3])}
\end{align}$$

So we're going to do this for $T$ periods, and we'll end up with:

$$ P_0 = \frac{\mathbb E[P_T]}{\prod_{t=1}^T (1 + \mathbb E[r_t])} + \sum_{t=1}^T \frac{X}{\prod_{s=0}^t(1 + \mathbb E[r_{t+s}])}$$

Assume that expected returns are constant -- that is, the distribution of returns doesn't change over time. Call this return $\mu$, we will have:

$$ P_0 = \frac{\mathbb E[P_T]}{(1 + \mu)^T} + \sum_{t=1}^T \frac{X}{(1 + \mu)^t}$$

Let's also assume that the expected return is positive. In this case, if we take the limit as $T \to \infty$, the first term will vanish bceause $(1 + \mu) > 1$, and $\lim_{t \to \infty} \frac{1}{x^t} = 0$ if $x > 1$. The second term will be a convergent geometric series with solution $\frac{\tilde X}{\mu} + \tilde X$. So in the limit, we find:

$$P_0 = \left ( \frac{1 + \mu}{\mu}\right ) \tilde X$$


# Exercise 9-10.3

## Question 1

We turn the CAPM into a regression problem by adding an ideosyncratic error term:

$$\tilde r_i = \alpha_i + \beta_i \tilde r_M + \varepsilon_t$$

How does this equation decompose the variance of $r_i$? 

Well, we want to know the variance, so apply the Variance operator to both sides of the equation:


$$\text{Var}(\tilde r_i) = \text{Var}(\alpha_i + \beta_i \tilde r_M + \varepsilon_t)$$

We're allowed to distribute the variance operation over the addition, but we have to multiply everything inside together first, because variances are quadratic. Also, all cross-terms become covariances. Basically, $Var(x, y) = Var(x^2) + Var(y^2) + 2 Cov(x, y)$. 

In our problem, $Cov(\tilde r_M, \varepsilon_t) = 0$ by assumption (because it's a regression problem).

$$
\begin{align}
\text{Var}(\tilde r_i) &= \text{Var}(\alpha_i^2) + \text{Var}(\beta_i \tilde r_M)^2 + \text{Var}(\varepsilon_t^2) \\
\sigma_i^2 &= 0 + \beta_i^2 \sigma_M^2 + \sigma_\varepsilon^2 \\
\sigma_i^2 &= \beta_i^2 \sigma_M^2 + \sigma_\varepsilon^2
\end{align}$$

We see that the variance of asset is equal to $\beta^2$ times the variance of the market, plus ideosyncratic variance $\sigma_\varepsilon^2$. Again, from the definition of $\beta$, we can write:

$$\beta_i = \frac{\sigma_{i,m}}{\sigma^2_m} = \frac{\rho \sigma_i \sigma_m}{\sigma^2_m} = \frac{\rho \sigma_i}{\sigma_m}$$

So $\beta^2_i = \frac{\rho^2 \sigma_i^2}{\sigma^2_m}$, which implies $\beta^2_i \sigma^2_m = \rho^2 \sigma_i^2$. This is exactly what appears in our variance decomposition, and it's the **systematic risk** in the asset.

The remaining risk, the **ideosyncratic** or **undiversifiable** risk, can be obtained by solving for $\sigma_\varepsilon^2$:

$$
\begin{align}
\sigma_i^2 &= \beta_i^2 \sigma_M^2 + \sigma_\varepsilon^2 \\
\sigma_i^2 &= \rho^2 \sigma_i^2 + \sigma_\varepsilon^2 \\
\sigma_\varepsilon^2 &= (1 - \rho^2) \sigma_i^2
\end{align}
$$

## Question 2

We actually already used the answer to this question in the previous question. To actually obtain the expression you need to minimize the sum of the errors

$$\min_{\alpha, \beta} \sum_t \varepsilon_t^2$$

With $\varepsilon_t = \tilde r_t - \alpha - \beta \tilde r_{M,t}$

I changed up the notation slightly to clarify the actual procedure. To estimate the market beta, we need to run a **time series regression** between the asset returns (from time t=0 to t=T) and market returns over the same period. This will give a single $\alpha, \beta$ pair for that asset. If we do one of these regressions for every asset in the market, we then talk about $\alpha_i, \beta_i$. But to actually get the coefficients, we need to first think about the time dimension, not the asset dimension.

So we take derivatives and set them equal to zero:

$$\frac{\partial}{\partial \alpha} = -2 \left (\sum_t \tilde r_t - \alpha - \beta \tilde r_{M,t})\right ) = 0$$

$$\frac{\partial}{\partial \beta} = -2 \left ( \sum_t \tilde r_{M,t} (\tilde r_t - \alpha - \beta \tilde r_{M,t}) \right ) = 0$$

Let's solve the first equation for $\alpha$. To begin, throw the -2 away, then distribute the sum over addition:

$$
\begin{align}\sum_t \tilde r_t - \sum_t \alpha - \sum_t \beta \tilde r_{M,t} &= 0 \\
T \mu_r - T\alpha - T \beta \mu_M &= 0 \\
\alpha = \mu_r - \beta \mu_M
\end{align}$$

To simplify, I used the fact that $\mu_x = \frac{1}{T}\sum_t x_i$. 

Next, we can plug this into the 2nd FoC. Throwing away the -2:

$$
\begin{align}
\left ( \sum_t \tilde r_{M,t} (\tilde r_t - \alpha - \beta \tilde r_{M,t}) \right ) &= 0 \\
\left ( \sum_t \tilde r_{M,t} (\tilde r_t - \mu_r + \beta \mu_M - \beta \tilde r_{M,t}) \right ) &= 0 \\
\left ( \sum_t \tilde r_{M,t} (\tilde r_t - \mu_r) - \beta r_{M,t} (\tilde r_{M,t} - \mu_M) \right ) &= 0 \\
\sum_t \tilde r_{M,t} (\tilde r_t - \mu_r) - \sum_t \beta r_{M,t} (\tilde r_{M,t} - \mu_M) &= 0
\end{align}$$

These are soooooo close to being something like covaraince expressions. If instead of $\tilde r_{M,t} (\tilde r_t - \mu_r)$ we had $(\tilde r_{M,t} - \mu_M)(\tilde r_t - \mu_r)$, we'd be there (after taking expectations).

So since that's what I want, I do that I have to do to get it. Subtract $\sum_t \mu_M(\tilde r_t - \mu_r)$ from both sides:

$$
\begin{align}
\sum_t \tilde r_{M,t} (\tilde r_t - \mu_r) - \sum_t \mu_M(\tilde r_t - \mu_r) - \sum_t \beta r_{M,t} (\tilde r_{M,t} - \mu_M) &= -\sum_t \mu_M(\tilde r_t - \mu_r) \\
\sum_t \left (\tilde r_{M,t} (\tilde r_t - \mu_r) - \mu_M(\tilde r_t - \mu_r) \right ) - \sum_t \beta r_{M,t} (\tilde r_{M,t} - \mu_M) &= -\sum_t \mu_M(\tilde r_t - \mu_r) \\
\sum_t (\tilde r_{M,t} - \mu_M)(\tilde r_t - \mu_r) - \sum_t \beta r_{M,t} (\tilde r_{M,t} - \mu_M) &= -\sum_t \mu_M(\tilde r_t - \mu_r) \\
\end{align} 
$$

Do the same "trick" with $\beta \sum_t \mu_M (\tilde r_{M,t} - \mu_M)$:

$$\sum_t (\tilde r_{M,t} - \mu_M)(\tilde r_t - \mu_r) - \sum_t \beta (r_{M,t} - \mu_M) (\tilde r_{M,t} - \mu_M) = -\sum_t \mu_M(\tilde r_t - \mu_r) + \beta \sum_t \mu_M (\tilde r_{M,t} - \mu_M)$$

This looks like a huge mess but it's about to get a lot better, we just have to do some work on the right hand side. Combine the sums and factor out $\mu_M$:

$$\mu_M \sum_t \left (-\tilde r_t + \mu_r - \beta \tilde r_{M,t} + \beta \mu_M \right ) $$

Distribute the sum over addition:

$$\mu_M \left (-\sum_t \tilde r_t + \sum_t \mu_r - \sum_t \beta \tilde r_{M,t} + \sum_t \beta \mu_M \right )$$

Use the mean identity we used above:

$$\mu_M \left (-T \mu_r + T \mu_r - \beta T \mu_M + \beta T \mu_M \right ) = 0$$

So we see that the right-hand side is in fact zero!

So we take expectations of what's left on the left-hand side:

$$
\begin{align}
\mathbb E \left [ \sum_t (\tilde r_{M,t} - \mu_M)(\tilde r_t - \mu_r) - \sum_t \beta (r_{M,t} - \mu_M) (\tilde r_{M,t} - \mu_M) \right ] &= 0 \\
\mathbb E \left [ \sum_t (\tilde r_{M,t} - \mu_M)(\tilde r_t - \mu_r) \right ] - \beta \mathbb E \left [\sum_t (r_{M,t} - \mu_M) (\tilde r_{M,t} - \mu_M) \right ] &= 0 \\
\sigma_{M, i} - \beta \sigma_{M}^2 = 0 \\
\beta = \frac{\sigma_{M, i}}{\sigma_M^2}
\end{align}
$$

The relationship is verified.

We can use this relationship to write an expression for the covariance between the asset and the market: $\sigma_{M, i} = \beta \sigma_M^2$

## Question 3

If we know $\alpha_i$ and $\beta_i$, we can obtain $\text{Cov}(\tilde r_i, \tilde r_j)$. Write down the CAPM regressions for both assets:

$$\begin{align}
\tilde r_i &= \alpha_i + \beta_i \tilde r_m + \varepsilon_i \\
\tilde r_j &= \alpha_j + \beta_j \tilde r_m + \varepsilon_j
\end{align}$$

Now we write down the covariance, substituting these expressions:

$$
\text{Cov}(\tilde r_i, \tilde r_j) = \text{Cov}(\alpha_i + \beta_i \tilde r_m + \varepsilon_i, \alpha_j + \beta_j \tilde r_m + \varepsilon_j) 
$$

Just like the variance, we can distribute out the covariance operator between all these linear terms. The variance (or covariance) of two deterministic terms will be zero, so things like $\text{Cov}(\alpha_i, \alpha_j) = 0$. One deterministic thing with a random thing is also zero, so $\text{Cov}(\alpha_i, \varepsilon_j) = 0$. We also have $\text{Cov}(\varepsilon_i, \varepsilon_j) = 0$, by definition. 

What will be left is:

$$\text{Cov}(\beta_i \tilde r_m, \beta_j \tilde r_m) = \beta_i \beta_j \text{Var}(r_m) = \beta_i \beta_j \sigma^2_M$$

So $\text{Cov}(\tilde r_i, \tilde r_j) = \beta_i \beta_j \sigma^2_M$

## Question 4

Consider a portfolio of two assets. This counts as a new asset, with $\sigma_p^2 = w^2 \sigma_1^2 + (1 - w)^2\sigma_2^2 + 2w(1-w)\sigma_{1,2}$

We know the ideosyncratic risk is given by $\sigma^2_\varepsilon = (1 - \rho^2)\sigma_p$. Substitute out expression:

$$\sigma^2_\varepsilon = (1 - \rho^2)\left ( w^2 \sigma_1^2 + (1 - w)^2\sigma_2^2 + 2w(1-w)\sigma_{1,2} \right )$$

If the weights are uniform, we have $w = (1 - w) = \frac{1}{2}$, because we have 2 assets, but in general we'll have $w = \frac{1}{N}$, and the ideosyncratic risk formula will be:

$$\sigma^2_\varepsilon = (1 - \rho^2)\left (\frac{1}{N^2} \sum_i \sigma_i^2 + \frac{2}{N^2} \sum_i \sum_{j\neq i} \sigma_{i,j} \right )$$

We know that as $N \to \infty$, $\frac{1}{N^2} \to 0$. Thus, $\lim_{N\to\infty} \sigma^2_\varepsilon = 0$. So we have shown that, as the number of assets in our portfolio grows, the ideosyncratic risk shrinks, leaving only the market risk.